Warning in added Lua BitOp during compiling on GCC 11.3

```
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c: In function ‘bit_band’:
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:101:5: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
  101 |     for (i = lua_gettop(L); i > 1; i--) b opr barg(L, i); BRET(b) }
      |     ^~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:102:1: note: in expansion of macro ‘BIT_OP’
  102 | BIT_OP(bit_band, &=)
      | ^~~~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:94:18: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
   94 | #define BRET(b)  lua_pushnumber(L, (lua_Number)(SBits)(b)); return 1;
      |                  ^~~~~~~~~~~~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:101:59: note: in expansion of macro ‘BRET’
  101 |     for (i = lua_gettop(L); i > 1; i--) b opr barg(L, i); BRET(b) }
      |                                                           ^~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:102:1: note: in expansion of macro ‘BIT_OP’
  102 | BIT_OP(bit_band, &=)
      | ^~~~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c: In function ‘bit_bor’:
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:101:5: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
  101 |     for (i = lua_gettop(L); i > 1; i--) b opr barg(L, i); BRET(b) }
      |     ^~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:103:1: note: in expansion of macro ‘BIT_OP’
  103 | BIT_OP(bit_bor, |=)
      | ^~~~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:94:18: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
   94 | #define BRET(b)  lua_pushnumber(L, (lua_Number)(SBits)(b)); return 1;
      |                  ^~~~~~~~~~~~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:101:59: note: in expansion of macro ‘BRET’
  101 |     for (i = lua_gettop(L); i > 1; i--) b opr barg(L, i); BRET(b) }
      |                                                           ^~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:103:1: note: in expansion of macro ‘BIT_OP’
  103 | BIT_OP(bit_bor, |=)
      | ^~~~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c: In function ‘bit_bxor’:
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:101:5: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
  101 |     for (i = lua_gettop(L); i > 1; i--) b opr barg(L, i); BRET(b) }
      |     ^~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:104:1: note: in expansion of macro ‘BIT_OP’
  104 | BIT_OP(bit_bxor, ^=)
      | ^~~~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:94:18: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘for’
   94 | #define BRET(b)  lua_pushnumber(L, (lua_Number)(SBits)(b)); return 1;
      |                  ^~~~~~~~~~~~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:101:59: note: in expansion of macro ‘BRET’
  101 |     for (i = lua_gettop(L); i > 1; i--) b opr barg(L, i); BRET(b) }
      |                                                           ^~~~
/home/tomi/fs2open.github.com/code/scripting/lua/bitop/bit.c:104:1: note: in expansion of macro ‘BIT_OP’
  104 | BIT_OP(bit_bxor, ^=)
      | ^~~~~~

```
#4274 added this. Probably should be silenced

--- a/code/CMakeLists.txt
+++ b/code/CMakeLists.txt
@@ -106,3 +106,6 @@ set_target_properties(code PROPERTIES XCODE_ATTRIBUTE_GCC_GENERATE_DEBUGGING_SYM
 set_target_properties(code PROPERTIES XCODE_ATTRIBUTE_COPY_PHASE_STRIP[variant=Debug] "NO")
 set_target_properties(code PROPERTIES XCODE_ATTRIBUTE_STRIP_INSTALLED_PRODUCT[variant=Debug] "NO")
 set_target_properties(code PROPERTIES XCODE_ATTRIBUTE_GCC_SYMBOLS_PRIVATE_EXTERN[variant=Debug] "NO")
+
+# Prevent GCC warnings in third-party BitOp... see GitHub #4366
+suppress_file_warnings(scripting/lua/bitop/bit.c)

